lib/metrics/peer_count.go

```diff
@@ -109,3 +109,103 @@ func (r *PeerConnFailure) Process(ctx context.Context, db *sql.DB, errs *common.
 func (r *PeerConnFailure) Clean(db *sql.DB, before int64) (int64, error) {
 	return common.Cleanup(db, "peerConnFailure", before)
 }
+
+type PeerCountByShard struct {
+	types.PeerCountByShard
+}
+
+type PeerCountByOrigin struct {
+	types.PeerCountByOrigin
+}
+
+func (r *PeerCountByShard) Process(ctx context.Context, db *sql.DB, errs *common.MetricErrors, data *types.TelemetryRequest) error {
+	if err := json.Unmarshal(*data.TelemetryData, &r); err != nil {
+		errs.Append(data.ID, fmt.Sprintf("Error decoding peer count by shard: %v", err))
+		return err
+	}
+
+	tx, err := db.BeginTx(ctx, nil)
+	if err != nil {
+		return err
+	}
+	defer tx.Rollback()
```

Comment:
To avoid calling rollback even when there's nothing to rollback, what do you think of changing this line and the commit on line 155 to something like this:
```go

	defer func() {
		if err == nil {
			err = tx.Commit()
			return
		}
		_ = tx.Rollback()
	}()
```

Reply:
I believe regular defer is standard Go practice:
```
Defer the transaction’s rollback. If the transaction succeeds, it will be committed before the function exits, making the deferred rollback call a no-op. If the transaction fails it won’t be committed, meaning that the rollback will be called as the function exits.
```

https://go.dev/doc/database/execute-transactions

IIRC the rollback function first checks if the tx is done, and if so will just return